storage/isolation_manager.rs

```diff
@@ -142,19 +130,18 @@ impl IsolationManager {
 
     fn await_pending_status_commits(
         &self,
-        predecessor_sequence_number: &SequenceNumber,
-        predecessor_window: &Arc<TimelineWindow<TIMELINE_WINDOW_SIZE>>,
+        predecessor_sequence_number: SequenceNumber,
+        predecessor_window: &TimelineWindow<TIMELINE_WINDOW_SIZE>,
     ) -> bool {
         debug_assert!(!matches!(predecessor_window.get_status(predecessor_sequence_number), CommitStatus::Empty));
         loop {
-            let status = predecessor_window.get_status(predecessor_sequence_number);
-            match status {
-                CommitStatus::Empty => unreachable!("Illegal state - commit status cannot move from pending"),
+            match predecessor_window.get_status(predecessor_sequence_number) {
+                CommitStatus::Empty => unreachable!("Illegal state - commit status cannot move from pending to empty"),
                 CommitStatus::Pending(_) => {
                     // TODO: we can improve the spin lock with async/await
                     // Note we only expect to have long waits in long chains of overlapping transactions that would conflict
                     // could also do a little sleep in the spin lock, for example if the validating is still far away
-                    continue;
+                    std::hint::spin_loop();
```

Comment:
This is better than just going to the start of the loop, as it may emit an instructions that signals to the CPU that this is a spin loop.